public_server/public_server.php

```diff
@@ -68,109 +61,100 @@ function public_server_cron($a,$b) {
 				'source_photo' => $a->get_baseurl() . '/images/person-80.jpg',
 			]);
 
-			q("update user set expire_notification_sent = '%s' where uid = %d",
-				dbesc(DateTimeFormat::utcNow()),
-				intval($rr['uid'])
-			);
+			$fields = ['expire_notification_sent' => DateTimeFormat::utcNow()];
+			dba::update('user', $fields, ['uid' => $rr['uid']]);
 		}
 	}
 
-	$r = q("select * from user where account_expired = 1 and account_expires_on < UTC_TIMESTAMP() - INTERVAL 5 DAY and account_expires_on > '0000-00-00 00:00:00'");
-	if(count($r)) {
-		foreach($r as $rr) {
-			User::remove($rr['uid']);
+	$nologin = Config::get('public_server', 'nologin', false);
+	if ($nologin) {
+		$r = q("SELECT `uid` FROM `user` WHERE NOT `account_expired` AND login_date <= '%s' AND `register_date` < UTC_TIMESTAMP() - INTERVAL %d DAY AND `account_expires_on` <= '%s'",
```

Comment:
Formatting: please wrap all field names in backticks or none (but I'd prefer all).